xla/backends/gpu/runtime/command_buffer_cmd_emitter.cc

```diff
@@ -334,8 +335,40 @@ static absl::Status AppendCommands(CommandBufferCmdSequence& cmd_sequence,
 static absl::Status AppendCommands(CommandBufferCmdSequence& cmd_sequence,
                                    const ThunkSequence& sequence,
                                    const ConvertToCommandsOptions& options) {
-  for (const std::unique_ptr<Thunk>& thunk : sequence)
+  absl::flat_hash_map<const Thunk*, CommandBufferCmd*> thunk_to_cmd;
+  for (const std::unique_ptr<Thunk>& thunk : sequence) {
+    int64_t previous_sequence_size = cmd_sequence.size();
     TF_RETURN_IF_ERROR(AppendCommands(cmd_sequence, *thunk, options));
+    if (cmd_sequence.size() > previous_sequence_size) {
+      thunk_to_cmd[thunk.get()] = cmd_sequence.back().get();
+    }
+  }
+
+  absl::flat_hash_map<CommandBufferCmd*, std::vector<CommandBufferCmd*>>
+      cmd_to_control_decessors;
```

Comment:
control_successors? (and everywhere below)

Reply:
fixed